lib/rules/prefer-const.js

```diff
@@ -300,7 +426,18 @@ module.exports = {
 
         return {
             "Program:exit"() {
-                groupByDestructuring(variables, ignoreReadBeforeAssign).forEach(checkGroup);
+                const identifierMap = groupByDestructuring(variables, ignoreReadBeforeAssign);
+
+                verifyAllDestructuring(identifierMap, checkingMixedDestructuring, destructureCount, destructureIdentifier).forEach(checkGroup);
+            },
+            "ExpressionStatement[expression.left.type = /ArrayPattern|ObjectPattern/]"(node) {
```

Comment:
Should this be `AssignmentExpression` rather than `ExpressionStatement`? Ostensibly we also want to catch assignments that aren't in an `ExpressionStatement`, e.g. `foo([bar] = 1)`.